base/strings/search.jl

```diff
@@ -1,20 +1,21 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
-function search(s::String, c::Char, i::Integer = 1)
+function findnext(pred::EqualTo{Char}, s::String, i::Integer)
     if i < 1 || i > sizeof(s)
         i == sizeof(s) + 1 && return 0
         throw(BoundsError(s, i))
     end
     @inbounds isvalid(s, i) || string_index_err(s, i)
-    c ≤ '\x7f' && return search(s, c % UInt8, i)
+    c = pred.x
+    c ≤ '\x7f' && return _search(s, c % UInt8, i)
     while true
-        i = search(s, first_utf8_byte(c), i)
+        i = _search(s, first_utf8_byte(c), i)
         (i == 0 || s[i] == c) && return i
         i = next(s, i)[2]
     end
 end
 
-function search(a::Union{String,ByteArray}, b::Union{Int8,UInt8}, i::Integer = 1)
+function _search(a::Union{String,ByteArray}, b::Union{Int8,UInt8}, i::Integer = 1)
```

Comment:
It looks like we need methods of `findfirst` and findnext` that call this (in order to keep using `memchr`)?

Reply:
Makes sense, though only the `search(::AbstractString, ::Chars)` method was actually documented (that's one of the problems with the current implementation: many methods are de facto exported though it's not clear whether they are public or not).

Reply:
I've added `findfirst`, `findnext`, `findlast` and `findprev` methods to look for an `Int8` or `UInt8` in a `ByteArray` to the first commit.